test/unit/cmap/connection.test.ts

```diff
@@ -137,6 +158,135 @@ describe('new Connection()', function () {
     });
   });
 
+  describe('#onMessage', function () {
+    context('when the connection is a monitoring connection', function () {
+      let queue: Map<number, OperationDescription>;
+      let driverSocket: FakeSocket;
+      let connection: Connection;
+
+      beforeEach(function () {
+        driverSocket = sinon.spy(new FakeSocket());
+      });
+
+      context('when multiple hellos exist on the stream', function () {
+        let callbackSpy;
+        const inputStream = new Readable();
+        const document = { ok: 1 };
+
+        beforeEach(function () {
+          callbackSpy = sinon.spy();
+          const firstHello = generateOpMsgBuffer(document);
+          const secondHello = generateOpMsgBuffer(document);
+          const thirdHello = generateOpMsgBuffer(document);
+          const buffer = Buffer.concat([firstHello, secondHello, thirdHello]);
+
+          connection = sinon.spy(new Connection(inputStream, connectionOptionsDefaults));
+          connection.isMonitoringConnection = true;
+          const queueSymbol = getSymbolFrom(connection, 'queue');
+          queue = connection[queueSymbol];
+
+          // Create the operation description.
+          const operationDescription: OperationDescription = {
+            requestId: 1,
+            cb: callbackSpy
+          };
+
+          // Stick an operation description in the queue.
+          queue.set(1, operationDescription);
+
+          // Push the buffer of 3 hellos to the input stream
+          inputStream.push(buffer);
+          inputStream.push(null);
+        });
+
+        it('calls the operation description callback with the document', async function () {
+          await on(inputStream, 'message');
```

Comment:
this would resolve immediately, no? iirc, `on` returns an `AsyncIterableIterator`

you can await a `once(...)` instead, but I think what you actually want is to set up the listener before you push things into the `inputStream`, and then await `onChangeIterator.next()` here; alternatively, since the current await does resolve sync, perhaps we don't need to await anything

Reply:
I needed to await something as the expectation was getting checked before the input stream piped to the message stream. I changed this to actually now add an extra assertion by awaiting `once` on the connection itself (since it does emit a message event as well on `onMessage` and asserting the first message has the expected `responseTo` value.